crypto/nrf_cc310_platform/src/nrf_cc3xx_platform_mutex_zephyr.c

```diff
@@ -172,6 +172,20 @@ nrf_cc3xx_platform_mutex_t heap_mutex = {
                     NRF_CC3XX_PLATFORM_MUTEX_MASK_IS_VALID
 };
 
+static bool mutex_flags_unknown(uint32_t flags){
+    switch(flags){
+        case (NRF_CC3XX_PLATFORM_MUTEX_MASK_IS_VALID | NRF_CC3XX_PLATFORM_MUTEX_MASK_IS_INTERNAL_MUTEX):
+        case (NRF_CC3XX_PLATFORM_MUTEX_MASK_IS_VALID | NRF_CC3XX_PLATFORM_MUTEX_MASK_IS_ALLOCATED):
+        case (NRF_CC3XX_PLATFORM_MUTEX_MASK_IS_VALID):
+        case NRF_CC3XX_PLATFORM_MUTEX_MASK_INVALID:
+        case NRF_CC3XX_PLATFORM_MUTEX_MASK_IS_HW_MUTEX:
+        case NRF_CC3XX_PLATFORM_MUTEX_MASK_IS_ATOMIC:
```

Comment:
This seems very arbitrary.

Where are the semantics defined?

You can be a valid internal mutex, and you can be a valid and allocated mutex, but you can't be a valid, allocated, and internal mutex?

Why can't the HW/atomic types be encoded with valided/allocated as well?

Reply:
@Vge0rge to comment on this

Reply:
The semantics of this are taken from the code of this file along with the code in the CryptoCell repository. The internal mutexes cannot be allocated because they are statically allocated inside the runtime library. You can see this in the mutex_init_platform library in this file. This functions check all the valid configuration of mutexes because the pointer to an allocate-able mutex can be come from the stack and is likely to have a random value there.

Reply:
"the pointer to an allocate-able mutex can be come from the stack and is likely to have a random value there."

So we are intentionally de-referencing uninitialized memory?

I don't think that's OK.

Is there any way to avoid it?

Reply:
We check the content of structure type that contains the mask and a pointer to a potentially allocatable mutex. (mutex->mutex). 

The pointer to the structure type itself verified to not be NULL here:
https://github.com/nrfconnect/sdk-nrfxlib/blob/424e68ba53601b087fb70d5efae32b9a6db97595/crypto/nrf_cc310_platform/src/nrf_cc3xx_platform_mutex_zephyr.c#L196

We don't intentionally dereference unitialized memory

Reply:
Code is overly complicated due to binary distribution. Not fixable.